src/Microsoft.ML.Parquet/ParquetLoader.cs

```diff
@@ -486,9 +503,18 @@ protected override bool MoveNextCore()
                         ds = ParquetReader.Read(_loader._parquetStream, _loader._parquetOptions, _readerOptions);
                     }
 
-                    int[] dataSetOrder = _rand == null ? Utils.GetIdentityPermutation(ds.RowCount) : Utils.GetRandomPermutation(_rand, ds.RowCount);
+                    IEnumerable<int> dataSetOrder;
+                    try
+                    {
+                        dataSetOrder = _rand == null ? Enumerable.Range(0, ds.RowCount) : Utils.GetRandomPermutation(_rand, ds.RowCount);
+                    }
+                    catch (OutOfMemoryException)
+                    {
+                        // if unable to create a shuffled sequence, default to sequential reading.
+                        dataSetOrder = Enumerable.Range(0, ds.RowCount);
+                    }
```

Comment:
This seems identical code with  line 402. Can this be factored out?

Reply:
Done.